src/Core/Output/TypedDataDumper.cs

```diff
@@ -41,11 +41,99 @@ public TypedDataDumper(EndianImageReader rdr, uint cbSize, Formatter stm)
         public void VisitArray(ArrayType at)
         {
             var addrEnd = rdr.Address + cbSize;
-            for (int i = 0; at.IsUnbounded || i < at.Length; ++i)
+
+            if (at.ElementType.Domain < Domain.Composite) 
```

Comment:
Relying on an enum's value in this way is fragile. Consider exposing  a `DataType.IsComposite` property _in a separate PR_` to accomplish this.